From 2357373a9050196dce19760580bc9441a8af1749 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 12 Aug 2018 11:50:07 +0300
Subject: [PATCH] btl: pcie: adjust the GEN3 registeres

1) disable the EQ
2) Enable direction of change
3) use presets 39F instead of 59F (remove 10 add 9)

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/btl/pcie_phy.s b/board/habanalabs/common/btl/pcie_phy.s
index 38ee3e5..dfedd0d 100644
--- a/board/habanalabs/common/btl/pcie_phy.s
+++ b/board/habanalabs/common/btl/pcie_phy.s
@@ -5,6 +5,7 @@
 #include "pcie_phy_regs.h"
 #include "pcie_aux_regs.h"
 #include "pcie_wrap_regs.h"
+#include "pcie_dbi_regs.h"
 
 #ifndef TARGET_IS_PLDM
 /*
@@ -35,6 +36,15 @@
 cfg mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G4_SLICE0, 0x00000009
 cfg mmPCIE_PHY_PROTOCOL0_EXT_TX_VBOOST_LVL, 0x00000004
 cfg mmPCIE_PHY_PHY_EXT_CTRL_SEL, 0x00000001
+
+/* disable the EQ */
+cfg mmPCIE_DBI_GEN3_RELATED_OFF, 0x00402801
+
+/*
+ * enable direction of change,
+ * use presets 39F instead of 59F (remove 10 add 9)
+ */
+cfg mmPCIE_DBI_GEN3_EQ_CONTROL_OFF, 0x05039F70
 #endif
 
 /* Clear phy_init (removes the workaround provided by SNPS) */
